src/context/AppProvider/AppContext.tsx

```diff
@@ -101,20 +102,13 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
    * handle wallet disconnect/switch logic
    */
   useEffect(() => {
-    // if we have a wallet and changed account ids, we have switched wallets
-    // NOTE! - the wallet will change before the account ids do, so clearing here is valid
-    // check the console logs in the browser for the ordering of actions to verify this logic
-    const switched = Boolean(wallet && requestedAccountIds.length)
-    const disconnected = !wallet
+    const walletId: WalletId | undefined = walletInfo?.deviceId
+    const switched = Boolean(walletId)
+    const disconnected = !walletId
     switched && moduleLogger.info('Wallet switched')
     disconnected && moduleLogger.info('Wallet disconnected')
-    if (switched || disconnected) {
-      dispatch(portfolio.actions.clear())
-      dispatch(txHistory.actions.clear())
-    }
-    // requestedAccountIds is changed by this effect
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [dispatch, wallet])
+    dispatch(portfolio.actions.setWalletId(walletId))
```

Comment:
starting to think this is an antipattern (reacting to an event from elsewhere) and the wallet provider should dispatch this? @gomesalexandre @0xApotheosis

Reply:
Discussed this one out and agree since this has only one dependency, might as well dispatch this wherever `walletId` update is originating from - right action wrong place